bsondump/bsondump.go

```diff
@@ -192,29 +218,30 @@ func (bd *BSONDump) Debug() (int, error) {
 func printBSON(raw bson.Raw, indentLevel int, out io.Writer) error {
 	indent := strings.Repeat("\t", indentLevel)
 	fmt.Fprintf(out, "%v--- new object ---\n", indent)
-	fmt.Fprintf(out, "%v\tsize : %v\n", indent, len(raw.Data))
+	fmt.Fprintf(out, "%v\tsize : %v\n", indent, len(raw))
 
-	//Convert raw into an array of RawD we can iterate over.
-	var rawD bson.RawD
-	err := bson.Unmarshal(raw.Data, &rawD)
+	elements, err := raw.Elements()
 	if err != nil {
 		return err
 	}
-	for _, rawElem := range rawD {
-		fmt.Fprintf(out, "%v\t\t%v\n", indent, rawElem.Name)
+	for _, rawElem := range elements {
+		key := rawElem.Key()
+		value := rawElem.Value()
+
+		fmt.Fprintf(out, "%v\t\t%v\n", indent, key)
 
 		// the size of an element is the combined size of the following:
 		// 1. 1 byte for the BSON type
 		// 2. 'e_name' : the BSON key, which is a null-terminated cstring
 		// 3. The BSON value
 		// So size == 1 [size of type byte] +  1 [null byte for cstring key] + len(bson key) + len(bson value)
 		// see http://bsonspec.org/spec.html for more details
-		fmt.Fprintf(out, "%v\t\t\ttype: %4v size: %v\n", indent, int8(rawElem.Value.Kind),
-			2+len(rawElem.Name)+len(rawElem.Value.Data))
+		fmt.Fprintf(out, "%v\t\t\ttype: %4v size: %v\n", indent, int8(value.Type),
```

Comment:
Each element is of type bson.RawElement, which is a []byte. For the length, can't we use len(element)?